### ISSUE TITLE
Inactive mobs when player is nearby

### ISSUE THREAD

**Issue by kugge** [PaperMC/Folia#42]
### Expected behavior

Normal mob behavior.

### Observed/Actual behavior

Sometimes, mobs are inactive when player is nearby. 
It seems to happen on chunk loading, but i'm not sure.
[Video](https://www.youtube.com/watch?v=i_aZN9N9nQs)

### Steps/models to reproduce

Step 1: Set time to night so enemies can spawn `/time set night`
Step 2: Set gamemode to survival `/gamemode survival`
Step 3: Invincibility, for testing purposes: `/effect give @s resistance infinite 255`
Step 4: TP far away: `/tp 100000 100 0`
Step 5: Walk till you encounter a hostile mob with no AI, repeat if needed

I can reproduce it after few tries.

### Plugin and Datapack List

None

### Folia version

```
> ver
[15:42:57 INFO]: This server is running Folia version git-Folia-"41c5548" (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 41c5548)
You are running the latest version
```

### Other

Here is everything i know:
- It does not always happen on generation. I saw this bug on old chunks
- Sometimes, spawning other mobs near freezed mobs also freeze them (? might be false. i can't reproduce it anymore)
- Hitting mob does not solve the problem.
- Logging out and logging back solve the problem.
- Might be related to the "inactive" mob feature.

**Comment by mani1232** [comment]
![image](https://user-images.githubusercontent.com/70480001/234780141-9d65b011-3338-4517-a4b5-b54e907bca3d.png)

In the red circle are inactive bees and only 1 bee is working

**Comment by Elavalur** [comment]
There is also a high chance fish would lose AI when entering above water ![image](https://user-images.githubusercontent.com/92751150/235383432-b4ff750f-1e1d-41f2-931a-86101361a603.jpeg)

**Comment by CozzyBro2** [comment]
I'm able to reproduce this very frequently, as it's happened many times today when my servers been up and we've played on it. Doesn't happen to every enemy mob, but if a couple pillagers during a raid will spawn, 1, 2, or rarely 3 of them will break when you move in their range.

No teleporting far away is needed to reproduce the issue as well, this happens just a couple hundred blocks from spawn.

Most importantly of all, I've discovered that this issue is not just their AI and Animations stopping, but also sometimes the enemy can move close to me, but will not use their weapon. For example, a creeper will get close to me, try to explode, and then break. Same with a vindicator, gets close but won't pull out its axe. I assume this is also why the pillagers break too, they want to begin the crossbow shoot sequence, but then they just dont.
